src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptionsSetup.cs

```diff
@@ -0,0 +1,102 @@
+            IEnumerable<ITelemetryInitializer> initializers,
+            IEnumerable<ITelemetryModule> modules)
+        {
+            this.applicationInsightsServiceOptions = applicationInsightsServiceOptions.Value;
+            this.initializers = initializers;
+            this.modules = modules;
+            this.telemetryChannel = serviceProvider.GetService<ITelemetryChannel>();
+        }
+
+        public void Configure(TelemetryConfiguration options)
+        {
+            if (this.applicationInsightsServiceOptions.InstrumentationKey != null)
+            {
+                options.InstrumentationKey = this.applicationInsightsServiceOptions.InstrumentationKey;
+            }
+
+            this.AddTelemetryChannelAndProcessorsForFullFramework(options);
+
+            options.TelemetryChannel = this.telemetryChannel ?? options.TelemetryChannel;
+
+            if (this.applicationInsightsServiceOptions.DeveloperMode != null)
+            {
+                options.TelemetryChannel.DeveloperMode = this.applicationInsightsServiceOptions.DeveloperMode;
+            }
+
+            if (this.applicationInsightsServiceOptions.EndpointAddress != null)
+            {
+                options.TelemetryChannel.EndpointAddress = this.applicationInsightsServiceOptions.EndpointAddress;
+            }
+
+            foreach (var initializer in this.initializers)
+            {
+                options.TelemetryInitializers.Add(initializer);
+            }
+
+            foreach (var module in this.modules)
+            {
+                module.Initialize(options);
+            }
+        }
+
+        private void AddTelemetryChannelAndProcessorsForFullFramework(TelemetryConfiguration configuration)
+        {
+#if NET451
+
+            // Adding Server Telemetry Channel if services doesn't have an existing channel
+            configuration.TelemetryChannel = this.telemetryChannel ?? new ServerTelemetryChannel();
+            if (configuration.TelemetryChannel is ServerTelemetryChannel)
+            {
+                // Enabling Quick Pulse Metric Stream
+                if (this.applicationInsightsServiceOptions.EnableQuickPulseMetricStream)
+                {
+                    var quickPulseModule = new QuickPulseTelemetryModule();
+                    quickPulseModule.Initialize(configuration);
+
+                    QuickPulseTelemetryProcessor processor = null;
+                    configuration.TelemetryProcessorChainBuilder.Use((next) => {
+                        processor = new QuickPulseTelemetryProcessor(next);
+                        quickPulseModule.RegisterTelemetryProcessor(processor);
+                        return processor;
+                    });
+                }
+
+                // Enabling Adaptive Sampling and initializing server telemetry channel with configuration
+                if (configuration.TelemetryChannel.GetType() == typeof(ServerTelemetryChannel))
```

Comment:
Aren't you already in this block because of the check above for "configuration.TelemetryChannel is ServerTelemetryChannel"?  Why check the type again a different way?